### ISSUE TITLE
Adding custom where condition parameter to the collection.

### ISSUE THREAD

**Issue by alankarmore** [Chumper/Datatable#292]
I am not sure whether it is an issue with the package or not, but actually I want some help regarding the below by using this package.

I have render the datatable as you have guided in the document. I want to filters the records with the custom filter option that is the value of the database column which is not present in the rendered table.

Suppose I have users list and I need to filter the records depends on the login status of the users, but I have not shown the login status as column in table. So when user clicks on the "online" link I want that database users table /  the collection will query with where condition as 'login_status' = 'online' and will return the records in the datatable again. 

So as per the where condition if I got 3 records then only this 3 records must needs to show in the datatable, currently which is not showing. It is showing all the records. Here is my sample method:

The method which will call to get the collection.
    /**
     * Displaying the datatable for the users lists.
     * 
     * @return Datatable\Table
     */
    public function getDatatable()
    {
        $status = Input::get('status');
        $result = $this->getUsers($status);
        return Datatable::collection($result)
                          ->addColumn('sr', function($model){
                                    return '<input type="checkbox" class="css-checkbox" id="checkbox'.$model->id.'"  checked="checked"/>
										<label for="checkbox'.$model->id.'" name="checkbox'.$model->id.'_lbl" class="css-label lite-green-check"></label>';
                            })                
                          ->addColumn('name', function($model){
                                    return '<h2>'.$model->name.'</h2>
                                            <ul id="list-nav">
                                                <li><a href="'.route('backend.edituser',['id' => $model->id]).'">BEARBEITEN</a></li>
                                                <li><a href="#">LÖSCHEN</a></li>
                                                <li><a href="#">ANSCHAUEN</a></li>
                                                <li><a href="#">MEDIEN</a></li>
                                            </ul>';
                            })
                        ->addColumn(trans('backend.users.index_last_activity_at'), function($model){
                            return $model->id;
                        })    
                        ->addColumn(trans('backend.users.index_activity'), function($model){
                            return $model->id;
                        })    
                        ->addColumn(trans('backend.users.index_visits'), function($model){
                            return $model->id;
                        })    
                        ->addColumn(trans('backend.users.index_fans'), function($model){
                            return $model->fans;
                        })    
                        ->setRowId(function($row) {
                                    return $row->id;
                                })
                        ->searchColumns('name')
                        ->orderColumns('name')
                        ->make();
    }

    /**
     *  Get all Users
     * 
     * @return collection UserModel
     */
    public function getUsers($where = null)
    {
        $query = UserModel::select('users.*', \DB::raw('COUNT(uf.fan_id) as fans'))
                ->leftjoin('user_follower_fan as uf', function($join) {
                            $join->on('users.id', '=', 'uf.follower_id');
                        });
        if (!empty($where)) {
            switch ($where) {
                case 'online':
                    $query->where('users.login_status','=',DB::raw(1));
                break;
                case 'active':
                    $query->where('users.status','=',DB::raw('active'));
                break;
            }
        }                
        
        $result = $query->groupBy('users.id')->get();
        
        return $result;
    }

Please ignore the joins and all from the query as of now. Thanks in advanced.

**Comment by Chumper** [comment]
you can prefilter the data you pass to datatable which i would suggest you to do.
So just get your data with the online flag set and pass it to datatables.
Or did i miss your requirement?

**Comment by alankarmore** [comment]
Yes I have prefilter the data on the basis of the where condition, It is giving me the JSON format which is return by the database query. Now the returned collection has been assigned to the $result variable and if I am assigning this $result to the datatables collection as Datatable::collection($result) // other code included .. 

But still it is not rendering the table in the view.

**Comment by alankarmore** [comment]
Hi Chumper this issue has been resolved. I have use, 

   ` ->setCallbacks(
        'fnServerParams', 'function ( aoData ) {
            aoData.push( { "name": "more_data", "value": "my_value" } );
        }'
    )`

To add an extra parameter for sorting / filtering records on basis of extra parameter. 
You are really created a great plugin to work.. Thanks... :100:
